### PR TITLE
Workflow to test generated minimal package

### LINKED ISSUE #277: Test generated package
### Description

Add a few tests that create a new package and ensure that some things are working:

- Running the tests should work
- Building the docs should work
- Maybe linting (the second time)? Although the first pass is supposed to fix it so the second pass works. Maybe it is a good test to ensure it

### PR DESCRIPTION
**Pull request by suvayu**
## Related issues

Checks:
- [x] Running the tests should work
- [x] Building the docs should work
- [ ] Lint check of the generated package

Closes #277

## Checklist

- [x] I am following the [contributing guidelines](https://github.com/abelsiqueira/BestieTemplate.jl/blob/main/docs/src/90-contributing.md)
- [x] Tests are passing
- [x] Lint workflow is passing
- [x] Docs were updated and workflow is passing
- [ ] [CHANGELOG.md](https://github.com/abelsiqueira/BestieTemplate.jl/blob/main/CHANGELOG.md) was updated

### CODE REVIEW

**Comment by suvayu** [comment]
I think something is off with the CI: https://github.com/suvayu/COPIERTemplate.jl/actions

File: .github/workflows/TestGeneratedPkg.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          julia --project=. --eval '
+            using BestieTemplate, UUIDs;
+            t = Dict(
+              "PackageName" => "Guldasta",
+              "PackageUUID" => string(UUIDs.uuid4()),
+              "PackageOwner" => "Bagicha",
+              "AuthorName" => "Mali",
+              "AuthorEmail" => "mali@bagicha.site"
+            );
+            BestieTemplate.generate("tmp/guldasta.jl", t; defaults = true, quiet = true);
```
Comment (abelsiqueira) [inline .github/workflows/TestGeneratedPkg.yml:62]:
```suggestion
            BestieTemplate.generate("tmp/Guldasta.jl", t; defaults = true, quiet = true);
```

File: .github/workflows/TestGeneratedPkg.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+          '
+      - name: Test generate package
+        run: |
+          julia --project=tmp/guldasta.jl --eval 'using Pkg; Pkg.test()'
```
Comment (abelsiqueira) [inline .github/workflows/TestGeneratedPkg.yml:66]:
```suggestion
          julia --project=tmp/Guldasta.jl --eval 'using Pkg; Pkg.test()'
```

File: .github/workflows/TestGeneratedPkg.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      - name: Generation docs
+        run: |
+          julia --project=tmp/guldasta.jl --eval 'using Pkg; Pkg.add("Documenter")' \
```
Comment (abelsiqueira) [inline .github/workflows/TestGeneratedPkg.yml:69]:
```suggestion
          julia --project=tmp/Guldasta.jl --eval 'using Pkg; Pkg.add("Documenter")' \
```

File: .github/workflows/TestGeneratedPkg.yml [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            -L tmp/guldasta.jl/docs/make.jl
```
Comment (abelsiqueira) [inline .github/workflows/TestGeneratedPkg.yml:70]:
```suggestion
            -L tmp/Guldasta.jl/docs/make.jl
```

File: .github/workflows/TestGeneratedPkg.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+      - name: Check docs genereation
```
Comment (abelsiqueira) [inline .github/workflows/TestGeneratedPkg.yml:71]:
```suggestion
      - name: Check docs generation
```

File: .github/workflows/TestGeneratedPkg.yml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        run: |
+          while IFS= read -r -d '' f; do
+            f=${f/src/build};
+            case $f in
+              *index.md) f=${f/%.md/.html};;
+              *) f=${f/%.md//index.html};;
+            esac
+            [[ -e $f ]] || { echo $f missing; }
+          done < <(find tmp/guldasta.jl/docs/ -type f -name '*.md' -print0 ) | grep missing
```
Comment (abelsiqueira) [inline .github/workflows/TestGeneratedPkg.yml:80]:
```suggestion
          done < <(find tmp/Guldasta.jl/docs/ -type f -name '*.md' -print0 ) | grep missing
```

File: .github/workflows/TestGeneratedPkg.yml [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        run: |
+          julia --project=tmp/guldasta.jl --eval 'using Pkg; Pkg.test()'
+      - name: Generation docs
+        run: |
+          julia --project=tmp/guldasta.jl --eval 'using Pkg; Pkg.add("Documenter")' \
+            -L tmp/guldasta.jl/docs/make.jl
+      - name: Check docs genereation
+        run: |
+          while IFS= read -r -d '' f; do
+            f=${f/src/build};
```
Comment (abelsiqueira) [inline .github/workflows/TestGeneratedPkg.yml:74]:
```suggestion
            f=${f/src/build};
            # Converts path/file.md to path/file/index.html (index.md just changes to index.html) 
```

File: .github/workflows/TestGeneratedPkg.yml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+              "PackageOwner" => "Bagicha",
+              "AuthorName" => "Mali",
+              "AuthorEmail" => "mali@bagicha.site"
+            );
+            BestieTemplate.generate("tmp/guldasta.jl", t; defaults = true, quiet = true);
+          '
+      - name: Test generate package
+        run: |
+          julia --project=tmp/guldasta.jl --eval 'using Pkg; Pkg.test()'
+      - name: Generation docs
```
Comment (abelsiqueira) [inline .github/workflows/TestGeneratedPkg.yml:67]:
```suggestion
      - name: Generate docs
```

File: .github/workflows/TestGeneratedPkg.yml [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+            t = Dict(
+              "PackageName" => "Guldasta",
+              "PackageUUID" => string(UUIDs.uuid4()),
+              "PackageOwner" => "Bagicha",
+              "AuthorName" => "Mali",
+              "AuthorEmail" => "mali@bagicha.site"
+            );
+            BestieTemplate.generate("tmp/guldasta.jl", t; defaults = true, quiet = true);
+          '
+      - name: Test generate package
```
Comment (abelsiqueira) [inline .github/workflows/TestGeneratedPkg.yml:64]:
```suggestion
      - name: Test generated package
```

**Review by abelsiqueira — APPROVED** [review]
I've made suggestions on typos and added comments to the code so I'll just go ahead and merge these directly.

**Review by abelsiqueira — APPROVED** [review]

File: .github/workflows/TestGeneratedPkg.yml [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+          julia --project=tmp/Guldasta.jl --eval 'using Pkg; Pkg.test()'
+      - name: Generate docs
+        run: |
+          julia --project=tmp/Guldasta.jl --eval 'using Pkg; Pkg.add("Documenter")' \
+            -L tmp/Guldasta.jl/docs/make.jl
+      - name: Check docs generation
+        run: |
+          while IFS= read -r -d '' f; do
+            f=${f/src/build};
+            # Converts path/file.md to path/file/index.html (index.md just changes to index.html)
```
Comment (abelsiqueira) [inline .github/workflows/TestGeneratedPkg.yml:75]:
```suggestion
            # Converts path/file.md to path/file/index.html (index.md just changes to index.html)
```

**Review by abelsiqueira — APPROVED** [review]

**Pull request merged**